unix/Makefile

```diff
@@ -108,6 +114,19 @@ SRC_MOD += modjni.c
 endif
 
 # source files
+ifeq ($(TARGET), freedos)
+# no moduselect for freedos
+SRC_C = \
+	main.c \
+	gccollect.c \
+	unix_mphal.c \
+	input.c \
+	file.c \
+	modos.c \
+	alloc.c \
+	coverage.c \
+	$(SRC_MOD)
+else
```

Comment:
All other configurable modules are configured in mpconfigport.mk.  Perhaps it's best to copy that pattern (see eg how modtime.c is done with MICROPY_PY_TIME).